### DISCUSSION TITLE
Persisting State with DuckDB WASM

### DISCUSSION THREAD

**Discussion by sswatson** [duckdb/duckdb#9299] [Q&A] [answered]
I want to persist the state of a DuckDB database in browser local storage, and doing so requires serializing the database contents in some way. I'm curious if there are good solutions to this problem.

I tried doing a database export/import cycle, but I don't know how to serialize the contents of the target directory in the virtual file system because I don't know how to read all the file names. It appears that the methods of the database object don't include a method for that operation.

I suppose one could alternatively try to build SQL dump functionality. In my situation, the tables involved are unlikely to be very large, so this would be fine. But it seems like a big undertaking to try to do it correctly from scratch, and I haven't been able to find anything already built.

Any thoughts would be much appreciated.

**Comment by Tishj** [comment, marked as answer]
Hi, I think this question is best asked over at the duckdb-wasm repo :)

**Comment by sswatson** [comment]
Apparently so! https://github.com/duckdb/duckdb-wasm/discussions/1433

Thanks for the suggestion.
